ykrt_internal/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+use std::ffi::{c_void, CString};
+use std::panic::UnwindSafe;
+use std::os::raw::c_char;
+
+use ykcompile::CompiledTrace;
+use yktrace::sir::{SirTrace, SIR};
+use yktrace::tir::TirTrace;
+
+fn handle_panic<T>(f: impl FnOnce() -> T + UnwindSafe) -> T {
+    std::panic::catch_unwind(|| f()).unwrap_or_else(|_| {
+        // It is not safe for panics to cross through the "C" abi.
+        // FIXME use "C unwind" as abi once it is implemented.
+        std::process::abort();
+    })
+}
+
+/// The different ways by which we can collect a trace.
```

Comment:
This enum is duplicated. Is there a way to share it? Is `include!` a bad idea?

Reply:
If you ever wanted to publish it on crates.io, that won't work.

Reply:
Fair enough. We can live with the duplication.